landingzones/views.py

```diff
@@ -480,6 +520,111 @@ def form_valid(self, form):
         )
 
 
+class ZoneUpdateView(
+    LoginRequiredMixin,
+    InvestigationContextMixin,
+    ZoneModifyMixin,
+    UpdateView,
+):
+    """LandingZone update view"""
+
+    model = LandingZone
+    form_class = LandingZoneForm
+    slug_url_kwarg = 'landingzone'
+    slug_field = 'sodar_uuid'
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data(*args, **kwargs)
+        context['zone'] = LandingZone.objects.get(
+            sodar_uuid=self.kwargs['landingzone']
+        )
+        return context
+
+    def get_permission_required(self, user):
+        """Get custom permission for user"""
+        if self.request.user == user:
+            return 'landingzones.update_zone_own'
+        else:
+            return 'landingzones.update_zone_all'
+
+    def get(self, request, *args, **kwargs):
+        """Override get() to ensure the zone status"""
+        zone = LandingZone.objects.get(sodar_uuid=self.kwargs['landingzone'])
+        redirect_url = reverse(
+            'landingzones:list', kwargs={'project': zone.project.sodar_uuid}
+        )
+        # Check permissions
+        if not self.request.user.has_perm(
+            self.get_permission_required(zone.user), zone.project
+        ):
+            msg = 'You do not have permission to update this landing zone.'
+            messages.error(request, msg)
+            return redirect(redirect_url)
+
+        # Check status
+        if zone.status not in STATUS_ALLOW_UPDATE:
+            messages.error(
+                request,
+                'Unable to update a landing zone with the '
+                'status of "{}".'.format(zone.status),
+            )
+            return redirect(redirect_url)
+
+        return super().get(request, *args, **kwargs)
```

Comment:
Minor style thing: If we have one "hanging" line at the end of a method, it's usually better to eliminate the empty line before it. I'm trying to clean up my old code to do that to avoid unnecessary blank lines, although you can still see bad examples here and  there :)